planetiler-core/src/main/java/com/onthegomap/planetiler/expression/Expression.java

```diff
@@ -31,6 +34,19 @@ public interface Expression {
   String POLYGON_TYPE = "polygon";
   String RELATION_MEMBER_TYPE = "relation_member";
 
+  String STRING_DATATYPE = "string";
+  String BOOLEAN_DATATYPE = "boolean";
+  String DIRECTION_DATATYPE = "direction";
+  String LONG_DATATYPE = "long";
+
+  static final Map<String, BiFunction<SourceFeature, String, Object>> dataTypeGetter =
+    Collections.unmodifiableMap(Map.of(
```

Comment:
Also (on a side note since this code will go away) `Map.of` returns an immutable map so you don't need to wrap it with an `Collections.unmodifiableMap`

Reply:
One of many poorly-conceived attempts at patching the non-deterministic behavior!  😬